src/odemis/driver/test/xt_client_test.py

```diff
@@ -953,5 +962,192 @@ def test_beamlet_index_info(self):
+        # Currently the range of the value can be quite big due to different designs for microscopes.
+        self.assertGreaterEqual(mpp_rotation, - math.radians(90))
+        self.assertLessEqual(mpp_rotation, math.radians(90))
+
+        # Check if VA is read only
+        with self.assertRaises(NotSettableError):
+            self.scanner.multiprobeRotation.value = mpp_rotation
+
+    def test_aperture_index_VA(self):
+        current_value = self.scanner.apertureIndex.value
+
+        # Test if directly changing it via the VA works
+        for test_aperture_index in self.scanner.apertureIndex.range:
+            self.scanner.apertureIndex.value = test_aperture_index
+            self.assertEqual(test_aperture_index, self.microscope.get_aperture_index())
+
+        # Test if errors are produced when a value outside of the range is set.
+        with self.assertRaises(IndexError):
+            self.scanner.apertureIndex.value = 1.2 * self.scanner.apertureIndex.range[1]
+        self.assertEqual(test_aperture_index, self.microscope.get_aperture_index())
+
+        # Test if the value is automatically updated when the value is not changed via the VA
+        self.microscope.set_aperture_index(0)
+        time.sleep(6)
+        self.assertEqual(0, self.scanner.apertureIndex.value)
+
+        self.scanner.apertureIndex.value = current_value
+
+    def test_beamlet_index_VA(self):
+        current_value = self.scanner.beamletIndex.value
+
+        # Test if directly changing it via the VA works
+        for test_beamlet_index in self.scanner.beamletIndex.range:
+            self.scanner.beamletIndex.value = test_beamlet_index
+            self.assertEqual(test_beamlet_index, self.microscope.get_beamlet_index())
+
+        # Test if errors are produced when a value outside of the range is set.
+        with self.assertRaises(IndexError):
+            self.scanner.beamletIndex.value = tuple(int(2 * i) for i in self.scanner.beamletIndex.range[1])
+        self.assertEqual(test_beamlet_index, self.microscope.get_beamlet_index())
+
+        # Test if the value is automatically updated when the value is not changed via the VA
+        self.microscope.set_beamlet_index(self.scanner.beamletIndex.range[0])
+        time.sleep(6)
+        self.assertEqual(self.scanner.beamletIndex.range[0], self.scanner.beamletIndex.value)
+
+        self.scanner.beamletIndex.value = current_value
+        self.assertEqual(current_value, self.microscope.get_beamlet_index())
+        self.assertEqual(current_value, self.scanner.beamletIndex.value)
+
+    def test_multiprobe_mode_VA(self):
+        current_beam_mode = self.scanner.multiBeamMode.value
+        current_aperture_index = self.scanner.apertureIndex.value
+        current_beamlet_index = self.scanner.beamletIndex.value
+
+        for multi_beam_boolean in [True, False, True, False, True]:
+            self.scanner.multiBeamMode.value = multi_beam_boolean
+            time.sleep(6)
```

Comment:
Is it really necessary to change the boolean this often with a 6 second sleep in between? Including the sleep at the bottom of this test, there is 36 seconds of sleeping in this test.

Reply:
Yes it is unfortunolty, you want to also test if the change from True --> False and the other way around succeeds also if it is preceded by switch in beam mode via the VA. This did fail before.
I have added a comment.